src/PointEvalHandler.jl

```diff
@@ -89,54 +91,66 @@ function _get_cellcoords(points::AbstractVector{Vec{dim,T}}, grid::AbstractGrid,
-    return all(x -> x > -tol, x_local) && sum(x_local) - 1 < tol
+    return all(x -> x ≥ -tol, x_local) && sum(x_local) - 1 < tol
 end
 
+cellcenter(::Type{<:RefHypercube{dim}}, _::Type{T}) where {dim, T} = zero(Vec{dim, T})
+cellcenter(::Type{<:RefSimplex{dim}}, _::Type{T}) where {dim, T} = Vec{dim, T}((ntuple(d->1/3, dim)))
+
+_solve_helper(A::Tensor{2,dim}, b::Vec{dim}) where {dim} = inv(A) ⋅ b
+_solve_helper(A::SMatrix{idim, odim}, b::Vec{idim,T}) where {odim, idim, T} = Vec{odim,T}(pinv(A) * b)
+
+# find_local_coordinate(interpolation::IP, cell_coordinates::Vector{V}, global_coordinate::V, warn::Bool, linesearch_max_substeps::Int , max_iters::Int, tol_norm::T) where {dim, T, V <: Vec{dim, T}, ref_shape, IP <: Interpolation{ref_shape}}
+#
+# 
+#
 # See https://discourse.julialang.org/t/finding-the-value-of-a-field-at-a-spatial-location-in-juafem/38975/2
-# TODO: should we make iteration params optional keyword arguments?
-function find_local_coordinate(interpolation, cell_coordinates::Vector{V}, global_coordinate::V) where {dim, T, V <: Vec{dim, T}}
+function find_local_coordinate(interpolation::IP, cell_coordinates::Vector{V}, global_coordinate::V, warn::Bool, linesearch_max_substeps::Int , max_iters::Int, tol_norm::T) where {dim, T, V <: Vec{dim, T}, ref_shape, IP <: Interpolation{ref_shape}}
     n_basefuncs = getnbasefunctions(interpolation)
     @assert length(cell_coordinates) == n_basefuncs
-    local_guess = zero(V)
-    max_iters = 10
-    tol_norm = 1e-10
+    local_guess = cellcenter(ref_shape, T)
     converged = false
-    for _ in 1:max_iters
-        global_guess = zero(V)
-        J = zero(Tensor{2, dim, T})
-        # TODO batched eval after 764 is merged.
-        for j in 1:n_basefuncs
-            dNdξ, N = shape_gradient_and_value(interpolation, local_guess, j)
-            global_guess += N * cell_coordinates[j]
-            J += cell_coordinates[j] ⊗ dNdξ
-        end
+    for iter in 1:max_iters
+        # Check if still inside element
+        check_isoparametric_boundaries(ref_shape, local_guess, sqrt(tol_norm)) || break
+        # Setup J(ξ) and x(ξ)
+        mapping, global_guess = calculate_mapping_and_spatial_coordinate(interpolation, local_guess, cell_coordinates)
+        J = getjacobian(mapping)
+        # Check if converged
         residual = global_guess - global_coordinate
-        if norm(residual) <= tol_norm
+        best_residual_norm = norm(residual) # for line search below
+        if best_residual_norm ≤ tol_norm
             converged = true
             break
         end
-        local_guess -= inv(J) ⋅ residual
+        if calculate_detJ(J) ≤ 0.0
+            warn && @warn "det(J) negative! Aborting! $(calculate_detJ(J))" 
+            break
+        end
+        Δξ = _solve_helper(J, residual) # J \ b throws an error. TODO clean up when https://github.com/Ferrite-FEM/Tensors.jl/pull/188 is merged.
+        # Do line search if outside/on boundary
```

Comment:
```suggestion
        # Do line search if outside boundary
```
To be consistent with the change to `≥` in `check_isoparameteric_boundaries`?